### PR TITLE
EL-1054 part 2: Pull 90-day-completion rate

### PR DESCRIPTION
**Pull request by patrick-laa**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-1054)

## What changed and why

After some discussion, we would like to display in Geckoboard the completion rate not for all time but for the last 90 days. 

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

File: app/services/metrics/for_key_metric_dashboard_service.rb [hunk 1]
```diff
@@ -134,7 +134,9 @@ def checks_completed(range = nil)
       relevant_events(range).where(event_type: "page_view", page: "view_results").count("DISTINCT assessment_code")
     end
 
-    def completion_rate(range = nil)
+    # The 'all time' completion rate is actually the last-90-day completion rate, as for a % rate, ancient historical
+    # data is less relevant.
+    def completion_rate(range = 90.days.ago.beginning_of_day..Date.current)
```
Comment (MazOneTwoOne) [inline app/services/metrics/for_key_metric_dashboard_service.rb:139]: question - do we want to return `nil` if this validates as true `return if started.zero? ` or do we want to return `0`?

**Review by MazOneTwoOne — APPROVED** [review]
Approving with a question.

File: app/services/metrics/for_key_metric_dashboard_service.rb (hunk 1 above)
Comment (patrick-laa) [reply app/services/metrics/for_key_metric_dashboard_service.rb:139]: I think it should be nil. If no checks have been started, then the completion _rate_ isn't zero. I'd argue there is no completion rate - in the absence of any checks, it doesn't make sense as a concept. Zero_ means "none of the checks that were started were completed", while nil means "this metric doesn't apply".

Comment (MazOneTwoOne) [reply app/services/metrics/for_key_metric_dashboard_service.rb:139]: that makes sense.

**Pull request merged**